# Lecture 10- Spring 2025
# Villas-Boas

# Lecture 10 EEP 118 Spring 2025

Please work on this notebook as your daily assignment afer Lecture 10 




In [1]:
# Load the 'pacman' package
install.packages("pacman")
library(pacman) 
#packages to use load them now using the pacman "manager"
p_load(dplyr, readr) 
#Another great feature of p_load(): if you try to load a package that is not installed on your machine, p_load() install the package for you, rather than throwing an error. For instance, let’s install and load one final package named ggplot2. 
p_load(ggplot2)

#set scientific display off, thank you Roy
options(scipen=999)

# Loading packages
pacman::p_load(lfe, lmtest, haven, sandwich, tidyverse,psych)
# lfe for running fixed effects regression
# lmtest for displaying robust SE in output table
# haven for loading in dta files
# sandwich for producing robust Var-Cov matrix
# tidyverse for manipulating data and producing plots
# psych for using describe later on




Installing package into ‘/srv/r’
(as ‘lib’ is unspecified)

Installing package into ‘/srv/r’
(as ‘lib’ is unspecified)

also installing the dependencies ‘mnormt’, ‘GPArotation’



psych installed



In [2]:
#-------------------------------------------
#1. Read in data
#-------------------------------------------
my_data <- read_dta("data2024.dta")
head(my_data)

timestamp         went2class soccerfan correct1 correct2 correctboth
1 2/6/2024 10:53:19 yes        yes       1        1        1          
2 2/6/2024 11:01:25 yes        yes       1        0        0          
3 2/6/2024 11:11:24 yes        yes       1        1        1          
4 2/6/2024 11:11:28 yes        yes       1        0        0          
5 2/6/2024 11:11:52 yes        yes       1        0        0          
6 2/6/2024 11:11:53 yes        no        1        1        1          
  numberCorrect went2Class isSoccerFan
1 2             1          1          
2 1             1          1          
3 2             1          1          
4 1             1          1          
5 1             1          1          
6 2             1          0

Start of Lecture 10 in line 161 of Lecture10.R in Bcourses

Let us look at the number of correct answers 
by the respondent, given by the variable my_data$numberCorrect

How would you compute the average number of correct answers
in the sample?

In [3]:
mean(my_data$numberCorrect)


[1] 1.527778

Break up the sample in two. those that came and those that did not come to class




In [4]:
#filter data
#select coming to class subsample 
my_dataClass <- filter(my_data, went2Class == 1)
#filter data not come to class, only 7 observ
my_dataNotClass<-filter(my_data,went2Class==0)


What is the average in the sample number of correct responses for coming to class?

In [5]:
#what is the average number of correct answers for those coming to class Nbar_c
#Note that I am typing it between parentheses the command below so the answer is printed in the log output , 
#and no need to call the variable after that

(Nhat_c<-mean(my_dataClass$numberCorrect))


#answer is 1.574257

[1] 1.574257

What is the average in the sample number of correct responses for not coming to class?

In [6]:
#what is the average number of correct answers for those not coming to class Nbar_nc
#Note that I am typing it between parentheses the command below so the answer is printed in the log output , 
#and no need to call the variable after that

(Nhat_nc<-mean(my_dataNotClass$numberCorrect))


#answer is 0.85714

[1] 0.8571429

In [7]:

#get variance and se for number of correct answers for those coming to Class 
var_Nhat_c<-var(my_dataClass$numberCorrect)/nrow(my_dataClass)
(se_Nhat_c<-sqrt(var_Nhat_c))
#will print the se_Nhat_c because I put it between ( )  parentheses in the line above



[1] 0.04944551

In [8]:

#get variance and std error for number of correct answers for those not coming to class
var_Nhat_nc<-var(my_dataNotClass$numberCorrect)/nrow(my_dataNotClass)
(se_Nhat_nc<-sqrt(var_Nhat_nc))
#will print the se_Nhat_nc because I put it between ( )  parentheses in the line above


#now we can do hypothesis testing using the five steps

[1] 0.340068

# Test whether the average number of correct answers for those that came to class (N_c) is statistically equal to the average number of correct answers for those that did not (N_nc) at the 10% significant level in a two-sided alternative

Remember the 5 steps in Hypothesis testing


### Step 1 specify the null and the alternative



Answer : H0: N_c = N_nc    or N_c - N_nc = 0 or D=0

Ha: D not equal to zero


## step 2: construct the t statistic under the null

$t = \frac{(Dhat-D_{under null}) }{ se(Dhat)}$

where $D_{under null}=0$

 and $Dhat=Nhat_c - Nhat_{nc}$

$se(Dhat)=\sqrt{ var(Nhat_c)+var(Nhat_{nc}) }$



In [9]:
#t_DA = (Dhat-Dunder null) / se(Dhat)

#where Dunder null=0

#Dhat=Nhat_c - Nhat_nc

Dhat<-Nhat_c-Nhat_nc

#se(Dhat)=square root of [ var(Nhat_c)+var(Nhat_nc) ]

se_Dhat<-sqrt(var_Nhat_c+var_Nhat_nc)

#t of daily assignment t_DA

t_ <- Dhat / se_Dhat

t_




[1] 2.086796

## Step 3: given significance level 10% what is the critical value we know its 1.659 by going to the table or by getting it in the comand below.

t_critical10<- qt(0.05, 106, lower.tail=FALSE)
t_critical10


what about for significance level 5%?

#find it in the cell below

#answer is obtained by the commands typed (first to get, then to show in output)


t_criticalDA<- qt(0.025, 106, lower.tail=FALSE)

t_criticalDA

In [10]:
#two sided ten percent
t_critical10<- qt(0.05, 106, lower.tail=FALSE)
t_critical10

[1] 1.659356

In [11]:
#two sided 5 percent
t_critical5<- qt(0.025, 106, lower.tail=FALSE)
t_critical5

[1] 1.982597

## Step 4: compare the absolute value of your tDA with the absolute value of the tcritical value. what do you conclude in Step 5?

Answer step 4 : the absolute value of the t stat is 2.08 is greater than the absolute 
value of the critical t of 1.9825972617655


so we land in the rejection area of this two-sided test

Answer step 5: 
so we reject the null that the mean of number of correct answers for those coming to class is the same as the mean of the number of correct answers for those not coming to class, against the alternative of the means being different at the 5 percent significance level.


## based on this test, reject that it does not matter coming to class in terms of the Null that the number of correct answers in the population would have the same mean regardless of coming or not coming to class  !

So, do come to class :-w


Go to Lecture Slides to learn about how to do interpretation and hypothesis testing on the parameters from the linear regression model before working on the bottom part of this notebook.

# Regression analysis of Lecture 10

Let us have the model $𝑁𝑢𝑚𝑏𝑒𝑟𝐶𝑜𝑟𝑟𝑒𝑐𝑡_𝑖=\beta_1+\beta_2  \ 𝑤𝑒𝑛𝑡2𝐶𝑙𝑎𝑠𝑠_𝑖  +𝑢_𝑖 $  

In [12]:
#model estimation
regLect8_1<-lm(numberCorrect~went2Class, my_data)
summary(regLect8_1)


Call:
lm(formula = numberCorrect ~ went2Class, data = my_data)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.8571 -0.5743  0.4257  0.4257  1.1429 

Coefficients:
            Estimate Std. Error t value  Pr(>|t|)    
(Intercept)   0.8571     0.1996   4.295 0.0000388 ***
went2Class    0.7171     0.2064   3.475  0.000741 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.528 on 106 degrees of freedom
Multiple R-squared:  0.1023,	Adjusted R-squared:  0.0938 
F-statistic: 12.08 on 1 and 106 DF,  p-value: 0.0007413


What does the constant estimate $\hat{\beta_1} $ mean?
It is the estimated or predicted average number of correct answers for those not coming to class
when went2Class=0

its estimate  is  0.8571

In [13]:
#another way to get that is
mean(my_dataNotClass$numberCorrect)

#we confirm it is indeed 
#[1] 0.8571429

[1] 0.8571429

We estimate $\hat{\beta_2}$ of went2Class= 0.7171.

Sign:
 The positive sign is as expected from people coming to
class and understanding the material
oh yess!!!!



Significance:

The t is  3.4575 with a corresponding p- value of 0.000741. 
We reject the null hypothesis that coming to class does not
affect the number of correct answers in the OVB survey


#Size:
#Coming to class is associated with a 0.7171 increase in 
#predicted number of correct answers in the survey all else equal.

#since the baseline was 0.8571 of those not having come to class
#then the predicted average number of correct answers for those coming to 
#class (went2Class=1) given our estimated model is


0.8571+0.7171*1


In [14]:
0.8571+0.7171*1

[1] 1.5742

In [15]:

#which is 
#[1] 1.5742

#which is the the predicted average number of correct answers for those coming to class

mean(my_dataClass$numberCorrect)


[1] 1.574257

## Does liking soccer matter for correct number of answers after controlling for having come to class?


In [16]:
#does liking soccer matter for correct number of answers after controlling
#for having come to class?

regLect8_2<-lm(numberCorrect~went2Class+isSoccerFan, my_data)
summary(regLect8_2)





Call:
lm(formula = numberCorrect ~ went2Class + isSoccerFan, data = my_data)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.8649 -0.5616  0.4112  0.4384  1.1623 

Coefficients:
            Estimate Std. Error t value  Pr(>|t|)    
(Intercept)  0.86493    0.20259   4.269 0.0000431 ***
went2Class   0.72390    0.20885   3.466  0.000766 ***
isSoccerFan -0.02727    0.10291  -0.265  0.791542    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5303 on 105 degrees of freedom
Multiple R-squared:  0.1029,	Adjusted R-squared:  0.08578 
F-statistic:  6.02 on 2 and 105 DF,  p-value: 0.003349


The coefficient on "isSoccerFan" is negative and not significantly different from zero
since the t stat of null hypothesis that the true coefficient is zero is
-0.265 which lies in the cannot rejection area at any significance level

the pvalue is 0.791542

isSoccerFan -0.02727    0.10291  -0.265 0.791542

The R squared is 0.1029  

before, without soccerfan in the regression the Rsquared is 0.1023


So, soccer fan does not matter and does not improve the R squared much,
what we call an irrelevant variable to explaing the number of correct answers
in a OVB survey after the OVB material was lectured
MAKES sense

I asked that question to use as a "placebo" variable I would add to the model
to see whether coming to class matters in answering a survey correctly and
being a soccer fan matters (totally random question and unrelated to OVB material understanding I would think...)


# THE END LECTURE 10 AND DAILY ASSIGNMENT LECTURE 10